frontends/p4/simplifyDefUse.cpp

```diff
@@ -179,6 +332,24 @@ class FindUninitialized : public Inspector {
         return defs;
     }
 
+    void initHeaderParam(const StorageLocation* storage, bool value) {
+        if (auto struct_storage = storage->to<StructLocation>()) {
+            if (struct_storage->isHeader()) {
+                headerDefs->update(struct_storage, value);
+            } else if (struct_storage->isStruct()) {
+                for (auto f : struct_storage->fields())
+                    initHeaderParam(f, value);
+            }
+        }
+    }
+
+    // Called at the beginning of controls, parsers and functions
+    void initHeaderParams(const IR::ParameterList* parameters) {
+        for (auto p : parameters->parameters)
+            if (auto storage = definitions->storageMap->getStorage(p))
+                initHeaderParam(storage, p->direction != IR::Direction::Out);
```

Comment:
I guess you can't really have header constants (with IR::Direction::None), but perhaps you should exclude these too.